testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1578,18 +1548,12 @@ fn test_network_key_rotation() {
     // Setup an RPC endpoint so we can talk to the node
     let validator_account = node_config.validator_network.as_ref().unwrap().peer_id();
     let libra = (&libra_interfaces).first().unwrap();
-    let last_reconfig = libra.last_reconfiguration().unwrap();
 
     // Rotate the validator network key
-    let (transaction_context, new_network_key) =
+    let (txn_ctx, new_network_key) =
         op_tool.rotate_validator_network_key(&backend).unwrap();
-
-    // Ensure all nodes have been reconfigured
-    wait_for_all_nodes(&libra_interfaces, |libra| {
-        libra.last_reconfiguration().unwrap() > last_reconfig
-    });
-
-    verify_transaction_executed(op_tool, transaction_context);
+    let mut client = swarm.get_validator_client(0, None);
+    client.wait_for_transaction(txn_ctx.address, txn_ctx.sequence_number + 1).unwrap();
```

Comment:
Woo, love to see shrunk tests

Reply:
This is kind of a bummer, in the ideal world, the code wouldn't have taken this interesting yet divergent path. The existing cli tool isn't very modular, so we cannot just take arbitrary pieces of it :/. I do look forward to the point in time where we can embrace async in the TCB code, because the overhead of maintaining both sync and async isn't fun or good.